src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -358,6 +347,12 @@ public static object CreateFromName(string name, params object[] args)
                     {
                         retvalType = null;
                     }
+
+                    if (retvalType != null)
```

Comment:
This is critical changes that improves perf by about 4-5x.

Reply:
would it make sense to add this to appNameHT instead and then make DefaultNameHT regular dictionary?

Reply:
Good idea. I made a change. I see that @adamsitnik implemented a different fix for the issue. See https://github.com/dotnet/corefx/pull/40184
We might want to take his change instead of this one.